examples/kubernetes/kafka-oauth-single-authz-service-accounts.yaml

```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: oidc-reader
+  annotations:
+    kubernetes.io/description: |
+      A cluster role which allows access to the OpenID Connect endpoints on
+      the API Server.
+rules:
+  - nonResourceURLs: ["/.well-known/openid-configuration", "/openid/v1/jwks"]
+    verbs: ["get", "post"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: oidc-reader-binding
+  annotations:
+  kubernetes.io/description: |
+    A role binding which allows for anonymous access to those endpoints
+    from consuming applications.
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: oidc-reader
+subjects:
+  - apiGroup: rbac.authorization.k8s.io
+    kind: User
+    name: "system:anonymous"
```

Comment:
This should have the service account of the Kafka brokers I guess and not anonymous?

Reply:
Since `strimizi-kafka-oauth` isn't setting its Bearer token to the one projected into the container, you need to allow `system:anonymous` to access the endpoints needed to retrieve the OpenID Connect settings and the Json Web Key Sets.

Reply:
If that is not enabled for everyone by default, it means this is likely a security risk? Should it really be used here? Shouldn't you instead use the broker's service account?

Reply:
I guess the same applies to `system:serviceaccount:default:default` -> maybe you should use some better example SA then the default one.

I'm also not sure I follow the relation ship of this example to the PR here. How do they relate? You do not seem to use any new flag in the example to disable the `Accept` header. So while the Kubernetes example is interesting, I do not follow the relationship here.

Reply:
I'm setting the value of `oauth.include.accept.header` to `false` via `spec.kafka.jvmOptions.javaSystemProperties` on line 63.

From my perspective, the ODIC and JWKS endpdoints in a "normal" application are exposed over the public internet. If someone is in a position to make a web request from inside of the kubernetes cluster to these endpoints, they are also going to be able to access the token for the service account and make those requests with authentication. If we want to use the brokers service account, then we should back this example out of this pull request and I'll open a new one which includes a dedicated authenticator for Kubernetes service accounts.

Reply:
We'll need to add the `includeAcceptHeader` config option to `oauth` listener authentication and `keycloak` authorization when we integrate the next OAuth release into the Strimzi Operator (another PR). We can then improve the example Kafka CR.

Reply:
> I'm setting the value of oauth.include.accept.header to false via spec.kafka.jvmOptions.javaSystemProperties on line 63.

I missed that part. As mentioned by Marko, this will need to be added tot he options after the next OAuth release. Until then it is a bit misleading since it will not work with OAuth 0.13.

> From my perspective, the ODIC and JWKS endpdoints in a "normal" application are exposed over the public internet. If someone is in a position to make a web request from inside of the kubernetes cluster to these endpoints, they are also going to be able to access the token for the service account and make those requests with authentication. If we want to use the brokers service account, then we should back this example out of this pull request and I'll open a new one which includes a dedicated authenticator for Kubernetes service accounts.

It seems that from Kubernetes perspective it is not something what they thing should be public. Also, thing like this will discourage users from using it as it does not _look secure_. If this works on any Kubernetes cluster, I think this would be a great usecase. But I think we need to work out the details to make sure it uses the service account and does not connect over as anonymous. So my vote would be to move this to a separate PR and do this through the broker service account.

Reply:
That's fine by me. I'll back this change out and start on a dedicated Kubernetes service account configuration which supports using the pod's service account token to authenticate with the api server.